src/core/toga/widgets/canvas.py

```diff
@@ -1,20 +1,297 @@
+from contextlib import contextmanager
 from .base import Widget
+from math import pi
 
 
 class Canvas(Widget):
-    """"""
+    """Create new canvas
+
+    Args:
+        id (str):  An identifier for this widget.
+        style (:class:`colosseum.CSSNode`): An optional style object. If no
+            style is provided then a new one will be created for the widget.
+        factory (:obj:`module`): A python module that is capable to return a
+            implementation of this class with the same name. (optional &
+            normally not needed)
+    """
 
     def __init__(self, id=None, style=None, factory=None):
+        super().__init__(id=id, style=style, factory=factory)
+
+        # Create a platform specific implementation of Canvas
+        self._impl = self.factory.Canvas(interface=self)
+
+        self.rehint()
+
+    @contextmanager
+    def save_restore(self):
+        """Push contents to a stack, then restore to the previous state
+
+        Yields: None
+
         """
+        self._impl.save()
+        yield
+        self._impl.restore()
+
+    def draw(self, draw_func):
+        self._impl.draw(draw_func)
+
+    # Line Styles
+
+    def line_width(self, width=2.0):
+        """Set width of lines
 
         Args:
-            id (str):  An identifier for this widget.
-            style (:class:`colosseum.CSSNode`): An optional style object. If no style is provided then a
-                new one will be created for the widget.
-            factory (:obj:`module`): A python module that is capable to return a
-                implementation of this class with the same name. (optional & normally not needed)
+            width (float): line width
+
         """
-        super().__init__(id=id, style=style, factory=factory)
-        self._impl = self.factory.Canvas(interface=self)
+        self._impl.line_width(width)
 
-        self.rehint()
+    # Fill and Stroke Styles
+
+    def fill_style(self, color=None):
+        """Color to use inside shapes
+
+        Currently supports color, in the future could support gradient and
+        pattern. A named color or RGBA value must be passed, or default
+        to black.
+        Args:
+            color (str): CSS color value or in rgba(0, 0, 0, 1) format
+
+        """
+        self._impl.fill_style(color)
+
+    def stroke_style(self, color=None):
+        """Color to use for lines around shapes
+
+        Currently supports color, in the future could support gradient and
+        pattern. A named color or RGBA value must be passed, or default to
+        black.
+        Args:
+            color (str): CSS color value or in rgba(0, 0, 0, 1) format
+
+        """
+        self._impl.stroke_style(color)
+
+    # Paths
+
+    @contextmanager
+    def begin_close_path(self, x, y):
+        """Creates a new path and then closes it
```

Comment:
Another naming issue here; as a context manager, would it make more sense for this to be just `closed_path()`, so usage would be:
```
with canvas.closed_path(1,2):
    canvas.line(...)
    canvas.line(...)
```
?

Reply:
Makes sense, I renamed to closed_path. Thanks!